actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -36,40 +35,11 @@ module ImplicitRender
     def default_render(*args)
       if template_exists?(action_name.to_s, _prefixes, variants: request.variant)
         render(*args)
-      elsif any_templates?(action_name.to_s, _prefixes)
-        message = "#{self.class.name}\##{action_name} does not know how to respond " \
-          "to this request. There are other templates available for this controller " \
-          "action but none of them were suitable for this request.\n\n" \
-          "This usually happens when the client requested an unsupported format " \
-          "(e.g. requesting HTML content from a JSON endpoint or vice versa), but " \
-          "it might also be failing due to other constraints, such as locales or " \
-          "variants.\n"
-
-        if request.formats.any?
-          message << "\nRequested format(s): #{request.formats.join(", ")}"
-        end
-
-        if request.variant.any?
-          message << "\nRequested variant(s): #{request.variant.join(", ")}"
-        end
-
-        raise ActionController::UnknownFormat, message
-      elsif interactive_browser_request?
-        message = "You did not define any templates for #{self.class.name}\##{action_name}. " \
-          "This is not necessarily a problem (e.g. you might be building an API endpoint " \
-          "that does not require any templates), and the controller would usually respond " \
-          "with `head :no_content` for your convenience.\n\n" \
-          "However, you appear to have navigated here from an interactive browser request – " \
-          "such as by navigating to this URL directly, clicking on a link or submitting a form. " \
-          "Rendering a `head :no_content` in this case could have resulted in unexpected UI " \
-          "behavior in the browser.\n\n" \
-          "If you expected the `head :no_content` response, you do not need to take any " \
-          "actions – requests coming from an XHR (AJAX) request or other non-browser clients " \
-          "will receive the \"204 No Content\" response as expected.\n\n" \
-          "If you did not expect this behavior, you can resolve this error by adding a " \
-          "template for this controller action (usually `#{action_name}.html.erb`) or " \
-          "otherwise indicate the appropriate response in the action using `render`, " \
-          "`redirect_to`, `head`, etc.\n"
```

Comment:
Trimming is fair, but I don't think we want to ditch the "this might not be a problem" entirely. The "Network Inspector -> Open in New Tab" workflow will be confusing if it sounds like the actually-successful XHR response failed.